src/main/java/com/socialmediaraiser/twitter/TwitterClient.java

```diff
@@ -261,45 +244,29 @@ public void retweetTweet(String tweetId) {
         String next;
         List<ITweet> result = new ArrayList<>();
         do {
-            JsonNode response = this.getRequestHelper().executeGetRequestWithParameters(this.getUrlHelper().getSearchTweets30daysUrl(),parameters);
-            JsonNode responseArray = null;
-            try {
-                responseArray = OBJECT_MAPPER.readTree(response.get("results").toString());
-            } catch (IOException e) {
-                LOGGER.severe(e.getMessage());
-            }
-
-            if(response.size() > 0){
-                try {
-                    result.addAll(Arrays.asList(OBJECT_MAPPER.treeToValue(responseArray, TweetDTOv1[].class)));
-                } catch (JsonProcessingException e) {
-                    LOGGER.severe(e.getMessage());
-                }
-            } else{
-                LOGGER.severe(()->nullOrIdNotFoundError);
-            }
-
-            if(!response.has(NEXT)){
+            Optional<TweetSearchV1DTO> tweetSearchV1DTO = this.getRequestHelper().executeGetRequestWithParameters(
+                    this.getUrlHelper().getSearchTweets30daysUrl(),parameters, TweetSearchV1DTO.class);
```

Comment:
Yes on this case it's obvious that a local field is better.
I was hesitating about all the others constructed url too. 
Like having a field of this type 
```private String userUrl = api.twitter.com/followers/user_id={user_id}```
and then get it and use `String.replace` to put the userId I want instead of having a get method which construct the whole URL. What do you think ?